src/Command/SelfUpdateCommand.php

```diff
@@ -103,17 +118,26 @@ protected function configure(): void
         );
     }
 
-    protected function doExecute(): int
+    protected function prepare(InputInterface $input): void
     {
-        $cachePath        = $this->getCachePath();
-        $this->downloader = new FileDownloader($cachePath, $this->config->getAuth());
+        parent::prepare($input);
+
+        /** @psalm-suppress DocblockTypeContradiction */
+        if ($this->downloader === null) {
```

Comment:
Rather use `isset()` here for compatibility with PHP 7.4+ where we will use typed properties - this will fail with an error that an uninitialized property must not be accessed prior initialization otherwise.

Reply:
We have to initialize the property in the constructor anyway when using typed properties. So we have to use `DownloaderInterface|null` as property type that's why I think checking against null here is correct.

Reply:
No, you do not have to - having uninitialized properties is perfectly fine.
We do this already for properties only valid within an execution flow, there is no enforcing of this (I'm aware of), you just must check with `isset` instead of `=== null` as this behaves somewhat like an check for `=== undefined`.